### PR TITLE
fix #819 add roll back on for GlobalTransaction

### LINKED ISSUE #819: add roll back on for GlobalTransaction
## Why you need it?
 Is your feature request related to a problem? Please describe in details
 add roll back on the needed exception for GlobalTransaction ;

## How it could be?
A clear and concise description of what you want to happen. You can explain more about input of the feature, and output of it.
 the exception throw then roll back ,otherwise commit ;

### PR DESCRIPTION
**Pull request by github-ygy**
### Ⅰ. Describe what this PR did
add roll back on for GlobalTransaction
roll back when we throw the exception what we want .

### Ⅱ. Does this pull request fix one issue?

fix #819

### CODE REVIEW

File: spring/src/main/java/io/seata/spring/annotation/GlobalTransactionalInterceptor.java [hunk 1]
```diff
@@ -106,19 +110,35 @@ public Object execute() throws Throwable {
...
                 public String name() {
                     String name = globalTrxAnno.name();
                     if (!StringUtils.isNullOrEmpty(name)) {
                         return name;
                     }
                     return formatMethod(methodInvocation.getMethod());
                 }
+
+                @Override
+                public TransactionInfo getTransactionInfo(){
```
Comment (CoffeeLatte007) [inline spring/src/main/java/io/seata/spring/annotation/GlobalTransactionalInterceptor.java]: { white space

File: spring/src/main/java/io/seata/spring/annotation/GlobalTransactionalInterceptor.java [hunk 2]
```diff
@@ -106,19 +110,35 @@ public Object execute() throws Throwable {
...
+                    TransactionInfo transactionInfo=new TransactionInfo();
```
Comment (CoffeeLatte007) [inline spring/src/main/java/io/seata/spring/annotation/GlobalTransactionalInterceptor.java]: info = new white space

File: tm/src/main/java/io/seata/tm/api/transaction/RollbackRule.java [hunk 3]
```diff
@@ -0,0 +1,95 @@
+package io.seata.tm.api.transaction;
```
Comment (CoffeeLatte007) [inline tm/src/main/java/io/seata/tm/api/transaction/RollbackRule.java]: Under copyright

File: tm/src/main/java/io/seata/tm/api/TransactionalTemplate.java [hunk 4]
```diff
@@ -42,23 +44,20 @@
...
-    public Object execute(TransactionalExecutor business) throws TransactionalExecutor.ExecutionException {
-
+    public Object execute(TransactionalExecutor business) throws Throwable {
         // 1. get or create a transaction
         GlobalTransaction tx = GlobalTransactionContext.getCurrentOrCreate();
 
+        //1.1 get transactionInfo
+        TransactionInfo txInfo = business.getTransactionInfo();
+        if (txInfo == null) {
+            throw new ShouldNeverHappenException("transactionInfo is not exists");
```
Comment (xingfudeshi) [inline tm/src/main/java/io/seata/tm/api/TransactionalTemplate.java]: transactionInfo does not exist

File: spring/src/main/java/io/seata/spring/annotation/GlobalTransactionalInterceptor.java (hunk 1 above)
Comment (github-ygy) [reply spring/src/main/java/io/seata/spring/annotation/GlobalTransactionalInterceptor.java]: done

File: spring/src/main/java/io/seata/spring/annotation/GlobalTransactionalInterceptor.java (hunk 2 above)
Comment (github-ygy) [reply spring/src/main/java/io/seata/spring/annotation/GlobalTransactionalInterceptor.java]: done

File: tm/src/main/java/io/seata/tm/api/transaction/RollbackRule.java (hunk 3 above)
Comment (github-ygy) [reply tm/src/main/java/io/seata/tm/api/transaction/RollbackRule.java]: done

File: tm/src/main/java/io/seata/tm/api/TransactionalTemplate.java (hunk 4 above)
Comment (github-ygy) [reply tm/src/main/java/io/seata/tm/api/TransactionalTemplate.java]: done

**Pull request merged**